microsoft/testsuites/core/provisioning.py

```diff
@@ -268,6 +268,37 @@ def verify_stop_start_in_platform(
             is_restart=False,
         )
 
+    @TestCaseMetadata(
+        description="""
+        This case performs a reboot stress test on the node
+        and iterates smoke test 100 times.
+        The test steps are almost the same as `smoke_test`.
+        The reboot times is summarized after the test is run
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+            supported_features=[SerialConsole],
+        ),
+    )
+    def stress_reboot(self, log: Logger, node: RemoteNode, log_path: Path) -> None:
+        reboot_times = []
+        try:
+            for i in range(100):
+                log.info(f"Reboot stress iteration {i+1}/100")
+                elapsed = self._smoke_test(log, node, log_path, "stress_reboot")
+                reboot_times.append((i + 1, elapsed))
+
+        finally:
+            log.info(f"Number of iterations completed: {len(reboot_times)}")
```

Comment:
There is an assertion in later code, so it doesn't need to log how many times rebooted. If it's not 100, the test case will be failed.